exercises/nucleotide-count/nucleotide_count.mli

```diff
@@ -1,7 +1,10 @@
 open Core_kernel
 
-(* Count the number of times the nucleotide occurs in the string. *)
-val count : string -> char -> int
+(* Determine if an input represents a valid nucleotide. *)
+val is_nucleotide : char -> bool
+
+(* Count the number of times a nucleotide occurs in the string. *)
+val count_nucleotide : string -> char -> int Option.t
```

Comment:
This could also return a Result which would allow for a descriptive error. The exercise would also be more focussed on Result.

Reply:
How about fixing it to `(int, char) Result.t` here?